pakyow-core/lib/core/middleware/req_path_normalizer.rb

```diff
@@ -0,0 +1,24 @@
+module Pakyow
+  module Middleware
+    # Rack compatible middleware that normalize the path if contains '//',
+    # it replace '//' with '/' and issue a 301 redirect to the new path.
+    #
+    # @api public
+    class ReqPathNormalizer
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env)
+        if env['PATH_INFO'].include? '//'
+          # normalize the path and issue 301 redirect
+          env['PATH_INFO'].gsub!('//', '/')
```

Comment:
I agree, it would definitely be more efficient. I was concerned doing that because we pass the entire `env` to `CallContext`. However, your comment made me think through it again and it's probably *more* accurate to pass the original, unmodified `env` into `CallContext` and the new path to `redirect`. Otherwise it looks like you're redirecting to the same path the request was for, which is wrong.